python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -241,23 +208,30 @@ def _collectModels(self, kerasModelsBytesRDD):
         :param paramMaps: list of ParamMaps matching the maps in `kerasModelsRDD`
         :return: list of MLlib models
         """
-        transformers = []
-        for (param_map, model_bytes) in kerasModelsBytesRDD.collect():
+        # we can just use the ordering of rdd because they are in the same order as params map
+        for i, (param_map, model_bytes) in enumerate(kerasModelsBytesRDD.collect()):
             model_filename = kmutil.bytes_to_h5file(model_bytes)
-            transformers.append({
-                'paramMap': param_map,
-                'transformer': KerasImageFileTransformer(modelFile=model_filename)})
-
-        return transformers
+            yield i, self._copyValues(KerasImageFileTransformer(modelFile=model_filename))
```

Comment:
Just one concern here, it looks like PySpark expects `fitMultiple` to return a threadsafe iterable ([see fitMultiple doc here](https://github.com/apache/spark/blob/30fcdc0/python/pyspark/ml/base.py#L90)) & by default achieves this through its own (internal) threadsafe iterator implementation, [see here](https://github.com/apache/spark/blob/30fcdc0/python/pyspark/ml/base.py#L30). I forget if PySpark prescribes a way developers should ensure thread safety, we can ask Joseph.